sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java

```diff
@@ -230,109 +231,120 @@ private AnalyzeSentimentResult convertToAnalyzeSentimentResult(DocumentSentiment
-        if (sentenceAspects == null) {
+        // If include opinion mining indicator is false, the service return null for the target list.
+        final List<SentenceTarget> sentenceTargets = sentenceSentiment.getTargets();
+        if (sentenceTargets == null) {
             return null;
         }
         final List<MinedOpinion> minedOpinions = new ArrayList<>();
-        sentenceAspects.forEach(sentenceAspect -> {
-            final List<OpinionSentiment> opinionSentiments = new ArrayList<>();
-            sentenceAspect.getRelations().forEach(aspectRelation -> {
-                final AspectRelationType aspectRelationType = aspectRelation.getRelationType();
-                final String opinionPointer = aspectRelation.getRef();
-                if (AspectRelationType.OPINION == aspectRelationType) {
-                    opinionSentiments.add(toOpinionSentiment(
-                        findSentimentOpinion(opinionPointer, documentSentimentList)));
+        sentenceTargets.forEach(sentenceTarget -> {
+            final List<AssessmentSentiment> assessmentSentiments = new ArrayList<>();
+            sentenceTarget.getRelations().forEach(targetRelation -> {
+                final TargetRelationType targetRelationType = targetRelation.getRelationType();
+                final String opinionPointer = targetRelation.getRef();
+                if (TargetRelationType.ASSESSMENT == targetRelationType) {
+                    assessmentSentiments.add(toAssessmentSentiment(
+                        findSentimentAssessment(opinionPointer, documentSentimentList)));
                 }
             });
-            final AspectSentiment aspectSentiment = new AspectSentiment(sentenceAspect.getText(),
-                TextSentiment.fromString(sentenceAspect.getSentiment().toString()),
-                sentenceAspect.getOffset(),
-                toSentimentConfidenceScores(sentenceAspect.getConfidenceScores()));
-            AspectSentimentPropertiesHelper.setLength(aspectSentiment, sentenceAspect.getLength());
-            minedOpinions.add(new MinedOpinion(aspectSentiment, new IterableStream<>(opinionSentiments)));
+            final TargetSentiment targetSentiment = new TargetSentiment();
+            TargetSentimentPropertiesHelper.setText(targetSentiment, sentenceTarget.getText());
+            TargetSentimentPropertiesHelper.setSentiment(targetSentiment,
+                TextSentiment.fromString(sentenceTarget.getSentiment().toString()));
+            TargetSentimentPropertiesHelper.setConfidenceScores(targetSentiment,
+                toSentimentConfidenceScores(sentenceTarget.getConfidenceScores()));
+            TargetSentimentPropertiesHelper.setOffset(targetSentiment, sentenceTarget.getOffset());
+            TargetSentimentPropertiesHelper.setLength(targetSentiment, sentenceTarget.getLength());
+
+            final MinedOpinion minedOpinion = new MinedOpinion();
```

Comment:
`MinedOpinions should now be `Opinions` right?

Reply:
I thought about it, but i think `Opinions` is too general.